src/include/migraphx/value.hpp

```diff
@@ -139,6 +140,19 @@ struct value
     using const_pointer   = const value_type*;
     using array           = std::vector<value>;
     using object          = std::unordered_map<std::string, value>;
+    struct binary : std::vector<std::uint8_t>
+    {
+        using base = std::vector<std::uint8_t>;
+        binary() {}
+        template <class Container>
+        explicit binary(const Container& c) : base(c.begin(), c.end())
```

Comment:
should be restrict the sizeof(c) == 1 here? otherwise, we will lose some data for the conversion.